figures/pipeline/enrollment_metrics.py

```diff
@@ -0,0 +1,272 @@
+"""This module collects metrics for a course enrollment (learner/course pair)
+
+# Overview
+
+This is new code for Figures 0.3.10. Its purpose is to address pipeline
+performance by reducing calls to `CourseGradeFactory().read.
+
+Calls are reduced by checking if the learner has a `StudentModule` record
+modified after the most recent `LearnerCourseGradeMetrics` record `date_for`
+field
+
+
+# Logic
+
+```
+for each learner+course
```

Comment:
Great section! Would you mind adding references in the logic to which function I should take a look at e.g.

```suggestion
for each learner+course  # see `main` function
```

Reply:
Will do!

Reply:
Actually, doing this for each line will add more noise than value, making the logic harder to read.  I'll at least add in references about the core iteration and update check